### ISSUE TITLE
Not able to receive even 100Gr from the new pow test giver

### ISSUE THREAD

**Issue by raiym** [ton-blockchain/ton#300]
I pulled latest changes and created the script as suggested in the docs: https://test.ton.org/TestGrams-HOWTO.txt 

The script monitors if transaction was made by testgiver and requests new seed and complexity and re-runs the pow-miner utility. 
Running on 2.3 GHz Quad-Core Intel Core i7 (8 cores) and still not able to mine pow in time. Also tried on 3.2 GHz Xeon W (16 cores) but without the script. 

If computation power is not an issue, what else it could be?

**Comment by igroman787** [comment]
We tried to "mine". Here's what came out of it (two tries):
```
> ./pow-miner -vv -w28 -t100 Ef_XOpw1giZMDzhJohdIvE4LA2hlE9O1RUxqbQqsNeGG5WlW 203163446660627297023326793794993952442 66531930102417552679378384690524267067425019809342012047680389367 100000000000 kf9CxReRyaGj0vpSH0gRZkOAitm_yDHvgiMGtmvG-ZTirrMC mined.boc

[ expected required hashes for success: 1740398768817 ]
[ hashes computed: 2591680383 ]
[ speed: 2.56946e+07 hps ]
```
```
> ./pow-miner -vv -w28 -t3600 Ef_XOpw1giZMDzhJohdIvE4LA2hlE9O1RUxqbQqsNeGG5WlW 203163446660627297023326793794993952442 66531930102417552679378384690524267067425019809342012047680389367 100000000000 kf9CxReRyaGj0vpSH0gRZkOAitm_yDHvgiMGtmvG-ZTirrMC mined.boc

[ expected required hashes for success: 1740398768817 ]
[ hashes computed: 93141830281 ]
[ speed: 2.58662e+07 hps ]
```

As you can see from two attempts, using 28 server cores was unsuccessful. The long-term operation of the program also did not bring success (we went through 93/1740 hashes in an hour).
Hence, we can only hope for luck, which directly depends on the number of hashes sorted out. And those, in turn, depend on the power of the iron and on the operating time.

**Comment by raiym** [comment]
@igroman787 thanks for sharing the results

**Comment by igroman787** [comment]
Suddenly it will be useful, so let's also have a successful brute-force iteration of the hash:
```
> /usr/bin/ton/crypto/pow-miner -vv -w10 -t10800 Ef_XOpw1giZMDzhJohdIvE4LA2hlE9O1RUxqbQqsNeGG5WlW 290654758487907751949376233055479468292 719712481031621425209468393577880020575432575272700909504830238907 100000000000 kf-kkdY_B7p-77TLn2hUhM6QidWrrsl8FYWCIvBMpZKprBtN mined.boc

[ expected required hashes for success: 160886593312 ]
4D696E65005F182FF25690D2AACC203003DBE333046683B698EF945FF250723C0F73297A2A1A41E2F10D97670E37877C6A2F9F95E8A9DA565E15F3D5066D33E9967D56379C7C04E3B6DAAA11FE3377DAA8850FEE7858B111040D97670E37877C6A2F9F95E8A9DA565E15F3D5066D33E9967D56379C7C04E3B6
Saving 176 bytes of serialized external message into file `mined.boc`
[ hashes computed: 184353451563 ]
[ speed: 2.62258e+07 hps ]
```

I set a stopwatch, and when it was 2 hours and 6 minutes, the search was over. Therefore, I think that the search itself was completely completed in about 2 hours (on 10 cores of server hardware).
Although I do not understand why, in total, more hashes were enumerated than was originally required (184353451563 > 160886593312).

**Comment by raiym** [comment]
@igroman787 thank you for update. 

I was able to mine test grams, but it happens very rarely and requires large computational power (=money), so I stopped the tests for now. 

I don't think that it is possible to mine *in time* even 100 Grams on a personal desktop. Only out of pure luck. 

It is very interesting though why did Telegram decide to make PoW smart-contracts and give away test Grams. It raises many interesting questions. 

As we can observe some people successfully mine 100 and 100K frequently and on a regular basis. 

Since we confirmed that it requires non-trivial computational power it means that these Grams cost something and people who mined them should be interested in supporting the test-net even after T. drops remaining servers but it is unclear where it would lead. 

It is also interesting who is mining these test grams and why. 

Let's see what will happen after August 1st. Very interesting. 
Good luck, Telegram!
